examples/html5-parser.rs

```diff
@@ -0,0 +1,18 @@
+use gosub_engine::html5::parser::document::{Document, DocumentBuilder};
+use gosub_engine::html5::{
+    input_stream::{Encoding, InputStream},
+    parser::Html5Parser,
+};
+
+fn main() {
+    // Creates an input stream
+    let mut stream = InputStream::new();
+    stream.read_from_str("<p>Hello<b>world</b></p>", Some(Encoding::UTF8));
+
+    // Initialize a document and feed it together with the stream to the html5 parser
+    let document = DocumentBuilder::new_document();
+    let _ = Html5Parser::parse_document(&mut stream, Document::clone(&document), None);

```

Comment:
Can't you say `document.clone()` here?
```suggestion
    let _ = Html5Parser::parse_document(&mut stream, document.clone(), None);
```

Reply:
I think you can.. but @emwalker had something to say about that.. (i forgot what :p)

Reply:
It is because it is a shallow copy and not a deep copy, which means, the heap memory doesn't get copied but reused.

Reply:
would it make sense to somehow change the naming to reflect this? Or is this the common idiomatic approach?

Reply:
Yes, i think this initially comes from the Rc or Arc smart pointers. For those smart pointers it is common to use ::clone() and not .clone() but not for the String smart pointer somehow🤷. I'm not completely sure, how the Document type is defined in the code, but from the suggestion from Eric, it is probably some kind of smart pointer.

Reply:
The convention with `Rc` is to use `Rc::clone(&reference)` instead of `reference.clone()`.  It's purely a developer signal.  As @Sharktheone mentioned, it's to let the developer know that a shallow copy is being made rather than a deep copy.

That said, @blaumeise20 likely knows more about Rust than I do, so if he reads this and still thinks `document.clone()` is preferable, we should go with what he says.

Reply:
Oh I understand now why you did this, although the reason for this is that is that you often have something *inside* the `Rc` which is handled like there was no `Rc` around it, and thus it makes it clearer that the `Rc` itself is cloned, not the content.

From the docs:
> `Rc<T>`'s implementations of traits like `Clone` may also be called using fully qualified syntax. Some people prefer to use fully qualified syntax, while others prefer using method-call syntax.

In this case, since `DocumentHandle` is not a wrapper type but explicitly a handle to `Document`, it make more sense to me to use the function call. But I can totally understand your point of the shallow copy. Do whatever you want ;-)

Reply:
I'm just a Rust newbie, still learning the conventions.